src/matter/cluster/GeneralCommissioningCluster.ts

```diff
@@ -26,37 +26,52 @@ const CommissioningErrorT = { tlvType: TlvType.UnsignedInt } as Template<Commiss
 
 const BasicCommissioningInfoT = ObjectT({
     failSafeExpiryLengthSeconds: Field(0, UnsignedIntT),
+    maxCumulativeFailsafeSeconds: Field(1, UnsignedIntT),
 });
 
-const SuccessFailureReponseT = ObjectT({
+const CommissioningSuccessFailureResponseT = ObjectT({
     errorCode: Field(0, CommissioningErrorT),
     debugText: Field(1, StringT),
 });
-export type SuccessFailureReponse = JsType<typeof SuccessFailureReponseT>;
+export type CommissioningSuccessFailureResponse = JsType<typeof CommissioningSuccessFailureResponseT>;
 
 const ArmFailSafeRequestT = ObjectT({
     expiryLengthSeconds: Field(0, UnsignedIntT),
-    breadcrumbStep: Field(1, UnsignedIntT),
+    breadcrumbStep: Field(1, UnsignedLongT),
 });
 
 const SetRegulatoryConfigRequestT = ObjectT({
     config: Field(0, RegulatoryLocationTypeT),
     countryCode: Field(1, StringT),
-    breadcrumbStep: Field(2, UnsignedIntT),
+    breadcrumbStep: Field(2, UnsignedLongT),
 });
 
+/**
+ * This cluster is used to manage global aspects of the Commissioning flow.
+ */
 export const GeneralCommissioningCluster = Cluster(
     0x30,
     "General Commissioning",
     {
-        breadcrumb: Attribute(0, UnsignedIntT),
+        breadcrumb: WritableAttribute(0, UnsignedLongT, BigInt(0)), /* readAcl: view, writeAcl: administer */
         commissioningInfo: Attribute(1, BasicCommissioningInfoT),
         regulatoryConfig: Attribute(2, RegulatoryLocationTypeT),
         locationCapability: Attribute(3, RegulatoryLocationTypeT),
+        supportsConcurrentConnections: Attribute(4, BooleanT, true),
     },
     {
-        armFailSafe: Command(0, ArmFailSafeRequestT, 1, SuccessFailureReponseT),
-        updateRegulatoryConfig: Command(2, SetRegulatoryConfigRequestT, 3, SuccessFailureReponseT),
-        commissioningComplete: Command(4, NoArgumentsT, 5, SuccessFailureReponseT),
+        /**
+         * Arm the persistent fail-safe timer with an expiry time of now + ExpiryLengthSeconds using device clock
+         */
+        armFailSafe: Command(0, ArmFailSafeRequestT, 1, CommissioningSuccessFailureResponseT),
+        /**
+         * Set the regulatory configuration to be used during commissioning
+         */
+        updateRegulatoryConfig: Command(2, SetRegulatoryConfigRequestT, 3, CommissioningSuccessFailureResponseT),
+        /**
+         * Signals the Server that the Client has successfully completed all steps of Commissioning/Reconfiguration
```

Comment:
nit: maybe can be shorten into
Informs that all steps of Commissioning/Reconfiguration needed during the fail-safe period have been completed.
All commands are for the client to tell something the server...
Yes, I know that was the original comment in Matter codebase...

Reply:
This is rather the good question ... leaving it the same as official documents or adjust ... :-) Will think about

Reply:
I copied it from the Chip-Tool XML file out of the "description" tags, but Ok i will rephrase it